bee-api/bee-rest-api/src/endpoints/config.rs

```diff
@@ -152,24 +131,36 @@ impl RestApiConfigBuilder {
         let port = multi_addr
             .iter()
             .find_map(|x| if let Protocol::Tcp(port) = x { Some(port) } else { None })
-            .unwrap_or_else(|| panic!("Unsupported protocol"));
-        let public_routes: Box<[String]> = self
-            .public_routes
-            .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(|s| s.to_string()).collect())
-            .into_boxed_slice();
-        let allowed_ips: Box<[IpAddr]> = self
-            .allowed_ips
-            .unwrap_or_else(|| DEFAULT_ALLOWED_IPS.to_vec())
-            .into_boxed_slice();
+            .expect("unsupported protocol");
+
+        let jwt_salt = self.jwt_salt.unwrap_or_else(|| DEFAULT_JWT_SALT.to_string());
+
+        let public_routes = {
+            let routes = self
+                .public_routes
+                .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(|r| r.to_string()).collect());
+            RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<String>>())
+                .expect("invalid public route provided")
+        };
+
+        let protected_routes = {
+            let routes = self
+                .protected_routes
+                .unwrap_or_else(|| DEFAULT_PROTECTED_ROUTES.iter().map(|r| r.to_string()).collect());
+            RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<String>>())
```

Comment:
A few redundant map closures here I think. Isn't clippy complaining? :thinking: 
Although I think Rust can infer the type of Vec:
`routes.iter().map(route_to_regex).collect::<Vec<_>>()`

Reply:
> A few redundant map closures here I think. Isn't clippy complaining? 🤔
I tried to change it but seems like it's necessary (and maybe thats the reason clippy isn't complaining).
Changing here the closures requires changes to the `route_to_regex ` or requires additional conversions. So I guess we should keep it as it

Reply:
alright then :+1: